### PR TITLE
BREAKING CHANGE: Improves extending ImportHandlers and refactors the Metrics API

### PR DESCRIPTION
**Pull request by andyHa**

### CODE REVIEW

File: src/main/java/sirius/biz/analytics/scheduler/README.md [hunk 1]
```diff
@@ -20,4 +20,14 @@ per month and an additional **best effort** one which tries to execute the compu
 day in the current month, so that the relected values are up to date (as long as the system 
 isn't overloaded).
 
+For maintenance and troubleshooting some options are available:
+* The [AnalyticsCommand](AnalyticsCommand.java) can be used in the [System Console](https://localhost:9000/system/console)
+by calling **analytics**. This lists all active schedulers, their queue and last execution. It also permits
+to forcefully start a scheduler imedietally.
```
Comment (sabieber) [inline src/main/java/sirius/biz/analytics/scheduler/README.md:26]:
```suggestion
to forcefully start a scheduler immediately.
```

File: src/main/java/sirius/biz/analytics/scheduler/README.md [hunk 2]
```diff
@@ -20,4 +20,14 @@ per month and an additional **best effort** one which tries to execute the compu
...
+* The Logger **analytics** can be set to FINE (using the **logger**) command to observer when the framework is
+doing.
+* The [Cluster State](https://localhost:9000/system/cluster) and the [Load Info](https://localhost:9000/system/load)
+report the queue lenghts in whiche the tasks are scheduler and also list which server is charce of executing
```
Comment (sabieber) [inline src/main/java/sirius/biz/analytics/scheduler/README.md:30]:
```suggestion
report the queue lengths in which the tasks are scheduler and also list which server is in charge of executing
```

File: src/main/java/sirius/biz/analytics/scheduler/README.md [hunk 3]
```diff
@@ -20,4 +20,14 @@ per month and an additional **best effort** one which tries to execute the compu
 day in the current month, so that the relected values are up to date (as long as the system 
 isn't overloaded).
 
+For maintenance and troubleshooting some options are available:
+* The [AnalyticsCommand](AnalyticsCommand.java) can be used in the [System Console](https://localhost:9000/system/console)
+by calling **analytics**. This lists all active schedulers, their queue and last execution. It also permits
+to forcefully start a scheduler imedietally.
+* The Logger **analytics** can be set to FINE (using the **logger**) command to observer when the framework is
```
Comment (sabieber) [inline src/main/java/sirius/biz/analytics/scheduler/README.md:27]:
```suggestion
* The Logger **analytics** can be set to FINE (using the **logger** command) to observe what the framework is
```

File: src/main/java/sirius/biz/tenants/mongo/MongoUserAccountImportHandler.java [hunk 4]
```diff
@@ -60,6 +62,11 @@ protected MongoUserAccountImportHandler(Class<?> clazz, ImporterContext context)
         super(clazz, context);
     }
 
+    @Override
+    protected void collectExcactLookups(Consumer<List<Mapping>> filterListConsumer) {
+        super.collectExcactLookups(filterListConsumer);
```
Comment (sabieber) [inline src/main/java/sirius/biz/tenants/mongo/MongoUserAccountImportHandler.java:67]: typo: `collectExcactLookups`

File: src/main/java/sirius/biz/analytics/metrics/MetricQuery.java [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+            case DAILY:
+                return LocalDate.now().minusDays(1);
+            case MONTHLY:
+                return LocalDate.now().minusMonths(1);
+            case YEARLY:
+                return LocalDate.now().minusYears(1);
+            case FACT:
+                return date.minusYears(999);
+            default:
+                return date.plusYears(999);
```
Comment (sabieber) [inline src/main/java/sirius/biz/analytics/metrics/MetricQuery.java:266]: sure?

File: src/main/java/sirius/biz/analytics/scheduler/BaseEntityBatchEmitter.java [hunk 6]
```diff
@@ -27,6 +27,11 @@
  * <p>
  * These batches are described using JSON and can be evaluated into an iterator of entities using
  * {@link #evaluateBatch(JSONObject, Consumer, Consumer)}.
+ *
+ * @param <I> the id type of the entities being processed by this emitter
+ * @param <C> the constraint type to be applied on entities processed by this emitter
+ * @param <B> the entity type being processed by this emitter
+ * @param <Q> the query type used to retireve entities being processed by this emitter
```
Comment (mkeckmkeck) [inline src/main/java/sirius/biz/analytics/scheduler/BaseEntityBatchEmitter.java:34]:
```suggestion
 * @param <Q> the query type used to retrieve entities being processed by this emitter
```

File: src/main/java/sirius/biz/importer/format/ImportDictionary.java [hunk 7]
```diff
@@ -61,7 +60,7 @@
     private Map<String, FieldDefinition> fields = new LinkedHashMap<>();
     private Map<String, String> aliases = new LinkedHashMap<>();
     private List<String> mappingFunction;
-    private Function<String, FieldDefinition> computedFieldLookup;
+    private List<Function<String, FieldDefinition>> computedFieldLookup = new ArrayList<>();
```
Comment (mkeckmkeck) [inline src/main/java/sirius/biz/importer/format/ImportDictionary.java:63]: Rename to computedFieldLookups ?

File: src/main/java/sirius/biz/importer/BaseImportHandler.java [hunk 8]
```diff
@@ -369,18 +377,48 @@ protected FieldDefinition expandAliases(FieldDefinition field) {
...
+                                .map(this::expandAliases)
+                                .ifPresent(exportDictionary::addField);
+            }
+        }
 
         return exportDictionary;
     }
 
+    /**
+     * Resolves an field into a <tt>FieldDefinition</tt>.
```
Comment (idlira) [inline src/main/java/sirius/biz/importer/BaseImportHandler.java:400]:
```suggestion
     * Resolves n field into a <tt>FieldDefinition</tt>.
```

**Pull request merged**